button-merchant/src/test/java/com/usebutton/merchant/PostOrderTaskTest.java

```diff
@@ -80,4 +91,103 @@ public void execute_limitAdTrackingTrue_verifyNullAdvertisingId() throws Excepti
 
         verify(buttonApi).postOrder(eq(order), eq(applicationId), eq(sourceToken), (String) isNull());
     }
+
+    @Test
+    public void execute_httpStatusException_wasRateLimited_verifyMaxRetry() throws Exception {
+        HttpStatusException httpStatusException = mock(HttpStatusException.class);
+        when(httpStatusException.wasRateLimited()).thenReturn(true);
```

Comment:
Do we really need to mock the exception? The following gets the job done in one less line of code 😂 

```java
HttpStatusException httpStatusException = new HttpStatusException("", 429);
```

Reply:
I only did that so we don't have to necessarily worry about the implementation of the class but just return true on the `wasRateLimited` method instead

Reply:
This doesn't make it depend on the internal implementation of `HttpStatusException` just the public API.

I think we would be relatively safe as I don't expect the definition of HTTP status codes to change any time soon ;)